packages-node/providence-analytics/src/program/services/InputDataService.js

```diff
@@ -215,22 +242,30 @@ function multiGlobSync(patterns, { keepDirs = false, root } = {}) {
 class InputDataService {
   /**
    * Create an array of ProjectData
-   * @param {PathFromSystemRoot[]} projectPaths
+   * @param {PathFromSystemRoot | ProjectInputData []} projectPaths
    * @param {Partial<GatherFilesConfig>} gatherFilesConfig
    * @returns {ProjectInputData[]}
    */
   static createDataObject(projectPaths, gatherFilesConfig = {}) {
     /** @type {ProjectInputData[]} */
-    const inputData = projectPaths.map(projectPath => ({
-      project: /** @type {Project} */ ({
-        name: pathLib.basename(projectPath),
-        path: projectPath,
-      }),
-      entries: this.gatherFilesFromDir(projectPath, {
-        ...this.defaultGatherFilesConfig,
-        ...gatherFilesConfig,
-      }),
-    }));
+    const inputData = projectPaths.map(projectPathOrObj => {
+      if (typeof projectPathOrObj === 'object') {
```

Comment:
typeof null === 'object'

Reply:
Yeah, we could validate the input again then. It would happen if someone does not comply with the supported type (string or {})

Reply:
Will add an input validation later 👍